### PR TITLE
Add validate webhook

### PR DESCRIPTION
**Pull request by serngawy**
Signed-off-by: melserngawy <melserng@redhat.com>

### CODE REVIEW

File: pkg/webhook/v1alpha1/placemanifestwork_validating.go [hunk 1]
```diff
@@ -55,6 +55,12 @@ func (r *PlaceManifestWorkWebhook) validateRequest(newPlaceWork *workv1alpha1.Pl
 }
 
 func validatePlaceManifests(placeWork *workv1alpha1.PlaceManifestWork) error {
+	if placeWork.Spec.PlacementRef.Name == "" {
```
Comment (qiujian16) [inline pkg/webhook/v1alpha1/placemanifestwork_validating.go]: this should be validated by openapi schema here https://github.com/open-cluster-management-io/api/blob/main/work/v1alpha1/types.go#L81

File: pkg/webhook/v1alpha1/placemanifestwork_validating.go [hunk 2]
```diff
@@ -55,6 +55,12 @@ func (r *PlaceManifestWorkWebhook) validateRequest(newPlaceWork *workv1alpha1.Pl
...
+		return apierrors.NewBadRequest("Placement name should not be empty")
+	}
 
+	if len(placeWork.Spec.ManifestWorkTemplate.Workload.Manifests) == 0 {
+		return apierrors.NewBadRequest("ManifestWorkTemplate workload should not be empty")
+	}
```
Comment (qiujian16) [inline pkg/webhook/v1alpha1/placemanifestwork_validating.go]: you may also want to validate the payload in manifests by reusing the same func for manifestwork validation

File: pkg/webhook/v1alpha1/placemanifestwork_validating.go (hunk 1 above)
Comment (serngawy) [reply pkg/webhook/v1alpha1/placemanifestwork_validating.go]: do you mean remove this validation and do it in the api repo ?

Comment (qiujian16) [reply pkg/webhook/v1alpha1/placemanifestwork_validating.go]: no I mean you could have a marker like this https://github.com/open-cluster-management-io/api/blob/main/addon/v1alpha1/types_clustermanagementaddon.go#L118-L120, that is sufficient to validate if name is empty.

Comment (serngawy) [reply pkg/webhook/v1alpha1/placemanifestwork_validating.go]: okay, just confirm wt to do

Comment (serngawy) [reply pkg/webhook/v1alpha1/placemanifestwork_validating.go]: done

File: pkg/webhook/v1/manifestwork_validating.go [hunk 3]
```diff
@@ -76,16 +76,24 @@ func (r *ManifestWorkWebhook) validateRequest(newWork, oldWork *workv1.ManifestW
 }
 
 func validateManifests(work *workv1.ManifestWork) error {
+	return ValidateManifestWorkSpec(work.Spec)
+}
+
+func ValidateManifestWorkSpec(manifestWorkSpec workv1.ManifestWorkSpec) error {
```
Comment (qiujian16) [inline pkg/webhook/v1/manifestwork_validating.go]: nit the name of ValidateManifestWorkSpec is a bit confusing, we actually validate manifest in the spec.

**Review by qiujian16 — COMMENTED** [review]
just a nit comment on naming,

/approve

File: pkg/webhook/v1/manifestwork_validating.go (hunk 3 above)
Comment (serngawy) [reply pkg/webhook/v1/manifestwork_validating.go]: I name it after the given param manifestWorkSpec and there is a validateManifest func [Line 106](https://github.com/open-cluster-management-io/work/pull/182/files#diff-c05eb54e7b9381bee7827595cf0c481f6c38395bdafaccfb692136bfddba67aaR106)

Comment (qiujian16) [reply pkg/webhook/v1/manifestwork_validating.go]: I would remove validateManifests in line 78 and name this func as ValidateManifests. Will it be OK?

Comment (serngawy) [reply pkg/webhook/v1/manifestwork_validating.go]: okay, done

**Pull request merged**